gossip/gossip_test.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestPullGossiperShutdown(t *testing.T) {
```

Comment:
Can we add a unit test that this will send a request and handle the response correctly?

Reply:
Test added here - https://github.com/ava-labs/coreth/blob/a360c286d7c3182cff4ad7a44b154bbea402dfe2/plugin/evm/tx_gossip_test.go

Reply:
I can't do it in the gossip package because it'll introduce a circular dependency, leaving it in the vm package for now.